src/main/resources/twilio-java/responseModel.mustache

```diff
@@ -35,10 +35,30 @@
     @JsonCreator
     private {{resourceName}}({{#vars}}
         @JsonProperty("{{{name}}}")
-        final {{{dataType}}} {{{name}}}{{^-last}},{{/-last}}{{/vars}}
+        {{#vendorExtensions.hydrate}}
+        final String {{{name}}}{{^-last}},{{/-last}}
+        {{/vendorExtensions.hydrate}}
+        {{#vendorExtensions.deserialize.x-currency}}
+        @JsonDeserialize(using = com.twilio.converter.CurrencyDeserializer.class)
+        {{/vendorExtensions.deserialize.x-currency}}
+        {{^vendorExtensions.hydrate}}
+        final {{{dataType}}} {{{name}}}{{^-last}},{{/-last}}
+        {{/vendorExtensions.hydrate}}
+    {{/vars}}
     ) {
     {{#vars}}
+        {{#vendorExtensions.hydrate.x-date-time-rfc-2822}}
+        this.{{{name}}} = {{{vendorExtensions.hydrate.x-date-time-rfc-2822}}};
+        {{/vendorExtensions.hydrate.x-date-time-rfc-2822}}
+        {{#vendorExtensions.hydrate.x-date-time}}
+        this.{{{name}}} = {{{vendorExtensions.hydrate.x-date-time}}};
+        {{/vendorExtensions.hydrate.x-date-time}}
+        {{#vendorExtensions.hydrate.x-date}}
+        this.{{{name}}} = {{{vendorExtensions.hydrate.x-date}}};
+        {{/vendorExtensions.hydrate.x-date}}
```

Comment:
Since this is the only place where the literal value stored in `vendorExtensions.hydrate` is used, can the value just be raised up a level? I.e., instead of storing the serializer string in `vendorExtensions.hydrate.x-date-time-rfc-2822`, can it be stored directly in `vendorExtensions.hydrate` so then this branching logic is not needed?